src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -122,47 +134,196 @@ class InlineTrans(Transformation):
+            defaults to None
+        :param check_unsupported_type: If `True`,
+            also perform checks (fail inlining) on arguments of
+            unsupported type, defaults to None
+        :param check_argument_unresolved_symbols: If `True`,
+            stop if encountering an unresolved symbol, defaults to None
+        """
+
+        self._call_routine_matcher.set_option(
+            ignore_missing_modules=ignore_missing_modules,
+            check_strict_array_datatype=check_argument_strict_array_datatype,
+            check_matching_arguments=check_argument_matching,
+            ignore_unresolved_types=check_argument_ignore_unresolved_types
+        )
+
+        if check_inline_codeblocks is not None:
+            self._option_check_codeblocks = check_inline_codeblocks
+
+        if check_diff_container_clashes is not None:
+            self._option_check_diff_container_clashes = (
+                check_diff_container_clashes)
+
+        if check_diff_container_clashes_unres_types is not None:
+            self._option_check_diff_container_clashes_unres_types = (
+                check_diff_container_clashes_unres_types
+            )
+
+        if check_resolve_imports is not None:
+            self._option_check_resolve_imports = check_resolve_imports
+
+        if check_static_interface is not None:
+            self._option_check_static_interface = check_static_interface
+
+        if check_array_type is not None:
+            self._option_check_array_type = check_array_type
+
+        if check_unsupported_type is not None:
+            self._option_check_unsupported_type = (
+                check_unsupported_type
+            )
+
+        if check_unresolved_symbols is not None:
+            self._option_check_unresolved_symbols = (
+                check_unresolved_symbols
+            )
+
+    def apply(
+        self, call_node: Call, routine_node: Routine = None, options=None
+    ):
+        """
         Takes the body of the routine that is the target of the supplied
         call and replaces the call with it.
 
-        :param node: target PSyIR node.
-        :type node: :py:class:`psyclone.psyir.nodes.Routine`
+        :param call_node: target PSyIR node.
+        :type call_node: :py:class:`psyclone.psyir.nodes.Call`
+        :param routine: PSyIR subroutine to be inlined.
+                Default: Automatically determine subroutine (search)
+        :type routine: :py:class:`psyclone.psyir.nodes.Routine`
         :param options: a dictionary with options for transformations.
         :type options: Optional[Dict[str, Any]]
         :param bool options["force"]: whether or not to permit the inlining
             of Routines containing CodeBlocks. Default is False.
-        '''
-        self.validate(node, options)
+        """
+
+        # Validate that the inlining can also be accomplish.
+        # This routine will also update
+        # self.node_routine and self._ret_arg_match_list
```

Comment:
Ah, so that's why there's state? Please update the comments on the declarations of these member variables.